src/jit/copyprop.cpp

```diff
@@ -468,6 +849,10 @@ void Compiler::optVnCopyProp()
 
         optBlockCopyProp(block, &curSsaName);
 
+        optCopyPropFoldCopyBlks(block);
```

Comment:
If you do not use `curSsaName` why do you need to put these in VN copy prop?

Reply:
Move it out to a simple block loop?

Reply:
I don't know. I'll have to take a close look at the problem you're trying to solve and figure out what might be a good solution, if any. My initial impression was that there are a bunch of `LCL_FLD`s involved and those are difficult to deal with in the current state of the JIT. 

The only thing that is trivial is the example I already mentioned, the one with `x2 = x0.f0;`. But fixing that shouldn't require all this complication, it might be something that local assertion propagation could handle.

Reply:
> it might be something that local assertion propagation could handle

Right -- seems like the cases we can easily handle (say an entire LCL_FLD source or dest for an unexposed local) would fit in pretty well in assertion prop.

Reply:
Happy for someone to try a different approach (or do this better); mostly am interested in the superfluous copies and the additional stack space the require going away,